compiler/ramallocation.go

```diff
@@ -0,0 +1,44 @@
+package compiler
+
+import (
+	"go/types"
+
+	"golang.org/x/tools/go/ssa"
+	"tinygo.org/x/go-llvm"
+)
+
+func (b *builder) createRAMAllocation(instr *ssa.CallCommon) error {
+	// Get the size, which must be a compile-time constant.
+	size, ok := instr.Args[0].(*ssa.Const)
```

Comment:
It could be really useful to be able to pass in a type and determine the size to allocate accordingly. 

This would let users define persistent variables; when their type definition changes, the reserved region grows/shrinks accordingly, without having to redo any arithmetic and update the `NewRAM` call separately (or risk forgetting to do so).

Just an idea, which may be a bit more elaborate than necessary for now.

Example use case:

```go
type Config struct {id int}

func main() {

	p := persistence.NewRAM(Config)

	// would be cool to either re-use the `make` builtin somehow, or create an equivalent:
	// p := persist(Config)
	// q := persist([]rune, 1024)
}
```

Reply:
Yeah - i think serialization needs to be explicit.

Reply:
How about the `type` argument in `NewRAM(type)` is required to implement a byte serialization interface? Something like the following:

```go
type RAMSerializer interface {
	Offset() int // zero unless its a subregion
	Size() int
	Serialize() ([]byte, error)
	Deserialize([]byte) error
}
```

This could facilitate higher-level `Get`/`Set` operations in the persistence API, making it friendlier to interact with; instead of _always_ having to juggle byte slices, the `ReadAt`/`WriteAt` methods could be invoked automatically internally via `RAMSerializer` interface.

Perhaps this functionality could just as well be implemented in user code... But if the pattern is useful and repeated often enough (which we cannot possibly know yet), why not eventually include it with the core API?